apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.YCSB/Program.cs

```diff
@@ -80,37 +95,109 @@ internal class Program : YcsbRunner
-                        });
-                    await cmd.ExecuteNonQueryAsync().ConfigureAwait(false);
+                            await UpdateOperationAsync().ConfigureAwait(false);
+                        }
+                    }
                     latencies.Add(sw.ElapsedMilliseconds);
                 }),
             new Operation("scanproportion", x => throw new NotImplementedException("scan is not yet implemented")),
             new Operation("insertproportion", x => throw new NotImplementedException("scan is not yet implemented"))
         };
 
+        private static Task UpdateOperationAsync()
+        {
+            var cmd = s_connection.CreateUpdateCommand(
+                s_table,
+                new SpannerParameterCollection
+                {
+                    {"id", SpannerDbType.String, s_keys[Rand.Value.Next(s_keys.Count)]},
+                    {s_fieldNames[Rand.Value.Next(10)], SpannerDbType.String, GenerateRandomString()}
+                });
+            return cmd.ExecuteNonQueryAsync();
+        }
+
+        private static Task RawUpdateOperationAsync() => RawStreamingReadOperationAsync();
+
+        private static async Task StreamingReadOperationAsync()
+        {
+            var cmd = s_connection.CreateSelectCommand(
+                s_selectQuery, new SpannerParameterCollection
+                {
+                    {"p", SpannerDbType.String, s_keys[Rand.Value.Next(s_keys.Count)]}
+                });
+            using (var reader = await cmd.ExecuteReaderAsync(s_timestampbound).ConfigureAwait(false))
+            {
+                while (await reader.ReadAsync().ConfigureAwait(false)) { }
+            }
+        }
+
+        private static V1.Type s_stringType = new V1.Type { Code = TypeCode.String};
+
+        private static async Task RawStreamingReadOperationAsync()
+        {
+            Session session;
+            s_rawsessions.TryDequeue(out session);
+            var stream = s_rawclient.ExecuteStreamingSql(
+                new ExecuteSqlRequest
+                {
+                    SessionAsSessionName = session.SessionName,
+                    Sql = s_selectQuery,
+                    Params = new Struct { Fields = { {"p", new Value { StringValue = s_keys[Rand.Value.Next(s_keys.Count)] }}}},
+                    ParamTypes = { {"p", s_stringType } },
+                    Transaction = new TransactionSelector
+                    {
+                        SingleUse = new TransactionOptions
+                        {
+                            ReadOnly = new TransactionOptions.Types.ReadOnly
+                            {
+                                Strong = true
+                            }
+                        }
+                    }
+                });
+            while (await stream.ResponseStream.MoveNext().ConfigureAwait(false))
+            {
+            }
+            stream.ResponseStream.Dispose();
+            s_rawsessions.Enqueue(session);
```

Comment:
this is just for testing but if something bad happens and we throw and never re-add the session to the queue, then we will nullref above.  This is ok for this test since it will actually make the test run much faster.... (and should show up in console err)